src/html/parse-frontmatter.js

```diff
@@ -136,188 +129,85 @@ const production = require('../utils/is-production');
-    // block) in order to warn about ambiguous nodes
-    lookahead(1, null),
-    // Filter out pairs in which both the start and the end is definately
-    // a settext heading or <hr>
-    reject(([fst, last]) => toIgnore(fst) && toIgnore(last)),
-    // Decide which blocks to ignore, which deserve warnings and which
-    // are actual frontmatter
-    procwarnigns,
-    filter(identity),
-    // Filter out false positive warnings for pseudo frontmatter blocks
-    // before actual frontmatter (warning gets invalidated by the fact
-    // that it DIRECTLY PRECEDES an actual frontmatter block)
-    lookahead(1, null),
-    reject(([val, next]) => true
-      && val.type === 'warning'
-      && val.warning.startsWith('Found ambigous frontmatter')
-      && next
-      && next.type === 'frontmatter'
-      && val.end === next.start),
-    /* eslint-disable-next-line no-unused-vars */
-    map(([val, next]) => val),
-  );
-};
-
-class FrontmatterParsingError extends Error {}
-
-const parseFrontmatter = ({ content = {} }, { logger }) => {
-  const { mdast, body } = content;
-
-  if (!mdast || !body) {
-    return;
-  }
-
-  // We splice the mdast.
-  let off = 0;
-
-  for (const block of list(findFrontmatter(mdast, body))) {
-    if (block.type === 'frontmatter') {
-      // Replace all the ast nodes making up a frontmatter block
-      // with the respective frontmatter block
-      const dat = {
-        type: 'yaml',
-        payload: block.payload,
-        position: {
-          start: cloneDeep(mdast.children[block.start + off].position.start),
-          end: cloneDeep(mdast.children[block.end + off].position.end),
-          indent: [],
-        },
-      };
-
-      const cnt = block.end - block.start + 1;
-      mdast.children.splice(block.start + off, cnt, dat);
-      off += -cnt + 1; // cnt removed, 1 inserted
-    } else {
-      const { warning, source, start } = block;
-      const fst = mdast.children[start + off];
-
-      // This also needs to account for settext headings
-      // hence the usage of end here, instead of using start
-      const { line } = fst.position.end;
-
-      // Source file pretty printing with line numbers
-      const sourceref = pipe(
-        source.split('\n'),
-        zipLeast2(range(line, Infinity)),
-        map(([l, no]) => `    ${no} | ${l} `),
-        join('\n'),
-      );
+    return eat(value.slice(0, offset + 3))({
+      type: 'yaml',
+      payload,
+    });
+  };
+}
 
-      const err = new FrontmatterParsingError(`${warning}\n${sourceref}`);
-      if (!production()) {
-        logger.warn(err);
-      }
-      logger.debug(err);
-    }
-  }
-};
+function plugin(opts) {
+  const { blockMethods, blockTokenizers } = this.Parser.prototype;
+  blockMethods.unshift('frontmatter');
+  blockTokenizers.frontmatter = create(opts);
+}
 
-assign(parseFrontmatter, { findFrontmatter, FrontmatterParsingError });
-module.exports = parseFrontmatter;
+module.exports = plugin;
```

Comment:
This needs to become a breaking change. If someone is constructing their own pipeline, and imports the `parseFrontmatter` step, it will break. Alternative: export a `nop` function that can still be imported, but that does nothing.

Reply:
yes...I thought of that, too.

Reply:
let's do both...